pkg/cloudproviders/aws/metadata.go

```diff
@@ -86,45 +79,45 @@ func GetMetadata(ctx context.Context) (*storage.ProviderMetadata, error) {
 	}, nil
 }
 
-func signedIdentityDoc(ctx context.Context, mdClient *ec2metadata.EC2Metadata) (*ec2metadata.EC2InstanceIdentityDocument, error) {
-	// This endpoint returns PKCS #7 structured data.
-	p7Base64, err := mdClient.GetDynamicDataWithContext(ctx, "instance-identity/rsa2048")
+func signedIdentityDoc(ctx context.Context, mdClient *imds.Client) (*imds.InstanceIdentityDocument, error) {
+	output, err := mdClient.GetDynamicData(ctx, &imds.GetDynamicDataInput{Path: "/instance-identity/rsa2048"})
 	if err != nil {
 		return nil, errors.Wrap(err, "retrieving RSA-2048 signature")
 	}
 
-	p7Raw, err := base64.StdEncoding.DecodeString(p7Base64)
+	p7Base64, err := io.ReadAll(output.Content)
```

Comment:
We can save one error with using read all at the end instead of the beginning 
```
	reader := base64.NewDecoder(base64.StdEncoding, output.Content)
	p7Raw, err := io.ReadAll(reader)
	if err != nil {
		return nil, errors.Wrap(err, "reading RSA-2048 signature")
	}
```